src/sys/windows/tcp.rs

```diff
@@ -290,6 +290,39 @@ pub(crate) fn get_keepalive(socket: TcpSocket) -> io::Result<bool> {
     }
 }
 
+pub(crate) fn set_tos(socket: TcpSocket, tos: u32) -> io::Result<()> {
+    match unsafe {
+        setsockopt(
+            socket,
+            IPPROTO_IP,
+            IP_TOS,
+            &tos as *const _ as *const c_char,
+            size_of::<c_ulong>() as c_int,

```

Comment:
This must be `size_of::<c_int>`. Also see https://github.com/rust-lang/socket2/blob/1c67209c2eda5708b3093900a59befbc4a981b13/src/socket.rs#L1171-L1205.